File: pkg/dev_compiler/test/codegen_expected/language/intrinsified_methods_test.js
dart_library.library('language/intrinsified_methods_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__intrinsified_methods_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const math = dart_sdk.math;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const intrinsified_methods_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  let intTonum = () => (intTonum = dart.constFn(dart.definiteFunctionType(core.num, [core.int])))();
  let VoidTovoid = () => (VoidTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [])))();
  intrinsified_methods_test.testIsNegative = function() {
    expect$.Expect.isFalse(12.0[dartx.isNegative]);
    expect$.Expect.isTrue((-12.0)[dartx.isNegative]);
    expect$.Expect.isFalse(core.double.NAN[dartx.isNegative]);
    expect$.Expect.isFalse(0.0[dartx.isNegative]);
    expect$.Expect.isTrue((-0.0)[dartx.isNegative]);
    expect$.Expect.isFalse(core.double.INFINITY[dartx.isNegative]);
    expect$.Expect.isTrue(core.double.NEGATIVE_INFINITY[dartx.isNegative]);
  };
  dart.fn(intrinsified_methods_test.testIsNegative, VoidTodynamic());
  intrinsified_methods_test.testIsNaN = function() {
    expect$.Expect.isFalse(1.0[dartx.isNaN]);
    expect$.Expect.isTrue(core.double.NAN[dartx.isNaN]);
  };
  dart.fn(intrinsified_methods_test.testIsNaN, VoidTodynamic());
  intrinsified_methods_test.testTrigonometric = function() {
    expect$.Expect.approxEquals(1.0, math.sin(math.PI / 2.0), 0.0001);
    expect$.Expect.approxEquals(1.0, math.cos(0), 0.0001);
    expect$.Expect.approxEquals(1.0, math.cos(0.0), 0.0001);
  };
  dart.fn(intrinsified_methods_test.testTrigonometric, VoidTodynamic());
  intrinsified_methods_test.foo = function(n) {
    let x = null;
    for (let i = 0; i <= dart.notNull(n); ++i) {
      expect$.Expect.equals(2.0, math.sqrt(4.0));
      intrinsified_methods_test.testIsNegative();
      intrinsified_methods_test.testIsNaN();
      intrinsified_methods_test.testTrigonometric();
    }
    return core.num._check(x);
  };
  dart.fn(intrinsified_methods_test.foo, intTonum());
  intrinsified_methods_test.main = function() {
    let m = intrinsified_methods_test.foo(4000);
  };
  dart.fn(intrinsified_methods_test.main, VoidTovoid());
  // Exports:
  exports.intrinsified_methods_test = intrinsified_methods_test;
});

File: tests/language_2/intrinsified_methods_test.dart
// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Dart test program for testing the instanceof operation.

library intrinsified_methods_test;

import "package:expect/expect.dart";
import 'dart:math';

testIsNegative() {
  Expect.isFalse((12.0).isNegative);
  Expect.isTrue((-12.0).isNegative);
  Expect.isFalse((double.nan).isNegative);
  Expect.isFalse((0.0).isNegative);
  Expect.isTrue((-0.0).isNegative);
  Expect.isFalse((double.infinity).isNegative);
  Expect.isTrue((double.negativeInfinity).isNegative);
}

testIsNaN() {
  Expect.isFalse((1.0).isNaN);
  Expect.isTrue((double.nan).isNaN);
}

testTrigonometric() {
  Expect.approxEquals(1.0, sin(PI / 2.0), 0.0001);
  Expect.approxEquals(1.0, cos(0), 0.0001);
  Expect.approxEquals(1.0, cos(0.0), 0.0001);
}

num foo(int n) {
  var x;
  for (var i = 0; i <= n; ++i) {
    Expect.equals(2.0, sqrt(4.0));
    testIsNegative();
    testIsNaN();
    testTrigonometric();
  }
  return x;
}

void main() {
  var m = foo(4000);
}
